pkg/operator/ceph/cluster/crash/reconcile.go

```diff
@@ -272,17 +277,36 @@ func (r *ReconcileNode) deleteCrashCollector(deployment appsv1.Deployment) error
 }
 
 func (r *ReconcileNode) reconcileCrashRetention(namespace string, cephCluster cephv1.CephCluster, cephVersion *cephver.CephVersion) error {
+
+	k8sVersion, err := k8sutil.GetK8SVersion(r.context.Clientset)
+	if err != nil {
+		return errors.Wrap(err, "failed to fetch get version")
+	}
 	if cephCluster.Spec.CrashCollector.DaysToRetain == 0 {
 		logger.Debug("deleting cronjob if it exists...")
-		cronJob := &v1beta1.CronJob{
-			ObjectMeta: metav1.ObjectMeta{
-				Name:      prunerName,
-				Namespace: namespace,
-			},
+
+		var errors error
+		minVersionForCronV1 := "1.21.0"
```

Comment:
suggesting to `minVersionForCronV1`  move this as a global constant as its being used in more than one places.